controller/src/beerocks/master/db/node.h

```diff
@@ -181,6 +199,7 @@ class node {
         std::unordered_map<int8_t, sVapElement> vaps_info;
     };
     std::shared_ptr<radio> hostap;
+    std::list<sTxRxLinkMetricsData> metrics_data_list;
```

Comment:
Since you'll need to update existing link metrics when new information is received from an agent, it would be more appropriate to use a map. Or a set (since the key is contained in the structure) but then you'll need to add an equality operator that only matches the al_mac addresses. Or it can be a map and remove the al_mac addresses from the link metrics struct (when iterating over the map, you get the keys anyway).

I'm inclined to go for the map.

Note that in principle, there should be an entry in the map for each combination of agent nodes (and you can get the list of agents with `get_all_connected_ires()`).